### PR TITLE
Implemented few async admin methods.

### PR DESCRIPTION
**Pull request by pawan-qlogic**
Implemented few methods. Please review and if any changes/suggestion, let me know, accordingly I will implement rest of the methods.

### CODE REVIEW

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 1]
```diff
@@ -277,19 +314,47 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
-  public CompletableFuture<Void> deleteSnapshot(String arg0) {
-    throw new UnsupportedOperationException("deleteSnapshot"); // TODO
-
+  public CompletableFuture<Void> deleteSnapshot(String snapshotName) {
+    return CompletableFuture.supplyAsync(() -> {
+    String btSnapshotName = null;
+    try {
+      btSnapshotName = getClusterName().toSnapshotName(snapshotName);
+    } catch (IOException e) {
+      FutureUtils.failedFuture(e);
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: I think that if you return `FutureUtils.failedFuture(e)` then you don't have to do `supplyAsync`

**Comment by pawan-qlogic** [comment]
Shall I go with the below approach ? Instead of
using FutureUtils.failedFuture(e), I am throwing CompletionException.

public CompletableFuture<Void> deleteSnapshot(String snapshotName) {
    return CompletableFuture.supplyAsync(() -> {
    String btSnapshotName = null;
    try {
      btSnapshotName = getClusterName().toSnapshotName(snapshotName);
    } catch (IOException e) {
      throw new CompletionException(e);
    }

    DeleteSnapshotRequest request = DeleteSnapshotRequest.newBuilder()
        .setName(btSnapshotName)
        .build();
    return request;
    }).thenCompose(
        d -> FutureUtils.toCompletableFuture(
            bigtableTableAdminClient.deleteSnapshotAsync(d)).thenApply(r ->
null));

  }

**Comment by sduskis** [comment]
@pawan-qlogic that works.  This might be more concise:

```java
public CompletableFuture<Void> deleteSnapshot(String snapshotName) {
    return CompletableFuture.supplyAsync(() -> {
    try {
      retrurn DeleteSnapshotRequest request = DeleteSnapshotRequest.newBuilder()
          .setName(getClusterName().toSnapshotName(snapshotName))
    } catch (IOException e) {
      throw new CompletionException(e);
    }

    }).thenCompose(
        d -> FutureUtils.toCompletableFuture(
            bigtableTableAdminClient.deleteSnapshotAsync(d)).thenApply(r ->
null));
  }
```

**Comment by sduskis** [comment]
FWIW, we still need tests for all of the new methods.

**Comment by sduskis** [comment]
Please add tests, and please don't implement any additional admin methods in this PR

**Comment by pawan-qlogic** [comment]
ok

**Comment by sduskis** [comment]
Can you please rebase instead of merge?  If you need help, please ask me or Alex.

**Comment by pawan-qlogic** [comment]
Done.

**Pull request closed without merging**